app/classes/custom_fields/data_types/number.rb

```diff
@@ -28,11 +31,25 @@ def coerce_value(value)
 
       private
 
+      def normalize_trailing_zeros(str)
+        str.gsub(TRAILING_ZEROS_REGEX, '\1').gsub(TRAILING_DOT_REGEX, '')
+      end
+
+      def valid_number_string?(str)
+        return false if str.empty?
+
+        coerced = coerce_value(str)
+        return false if coerced.nil?
+
+        normalize_trailing_zeros(str) == coerced.to_s
+      end
+
       def string_to_number(str)
-        if str.empty?
-          nil
-        elsif str =~ /\.\d+\z/
-          str.to_f
+        return nil if str.empty?
+
+        if str.include?('.')
+          normalized = normalize_trailing_zeros(str)
```

Comment:
Is there any use in deleting trailing zeros before converting to number? I don't think it will matter if we convert '1.20' or '1.2' to float